torchvision/datasets/forksafeziplookup.py

```diff
@@ -0,0 +1,46 @@
+import zipfile
+import threading
+import os, io
+import struct
+
+
+_FH_FILENAME_LENGTH = 10
+_FH_EXTRA_FIELD_LENGTH = 11
+STRUCT_FILE_HEADER = "<4s2B4HL2L2H"
```

Comment:
Where do this and the other constants come from?

Reply:
They come from Python's built-in `zipfile` module. Actually, it is cleaner to import them from there than to duplicate them. If Python ever changes the internals of the zipfile module (unlikely) we'll see then.